Corrected Get-adcomputer

sergioirium opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Corrected Get-adcomputer on "Remove-ConstructedState" and "Set-ConstructedState" functions.
resolved error: Unable to contact the server. This may be because this server does not exist, it is currently down, or it does not have the Active Directory Web Services running.
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
win_domain_computer.ps1

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

sergioirium committed:
Corrected Get-adcomputer

Corrected Get-adcomputer on "Remove-ConstructedState" and "Set-ConstructedState" functions.
resolved error: Unable to contact the server. This may be because this server does not exist, it is currently down, or it does not have the Active Directory Web Services running.

ShachafGoldstein commented on lib/ansible/modules/windows/win_domain_computer.ps1:
```diff
@@ -112,14 +112,14 @@ Function Set-ConstructedState($initial_state, $desired_state) {
     # Move computer to OU
     Try {
-      Get-ADComputer -Identity $desired_state.name |
+      Get-ADComputer -Identity $desired_state.name -credential $credential |
```
Should be @extra_args like on line 120

ShachafGoldstein commented on lib/ansible/modules/windows/win_domain_computer.ps1:
```diff
@@ -147,8 +147,8 @@ Function Add-ConstructedState($desired_state) {
   Try {
-    Get-ADComputer $initial_state.name `
-    | Remove-ADObject `
+    Get-ADComputer -identity $initial_state.name -credential $credential |
```
Should be @extra_args like on line 120

sergioirium replied:
I thought it was clearer to use the "-credential" parameter but if you think it is better "@extra_args" I change it. Thank you

ShachafGoldstein replied:
I think it is better since there may be other parameters such as the server one you want to also have for the fix

ShachafGoldstein commented on lib/ansible/modules/windows/win_domain_computer.ps1:
```diff
@@ -112,14 +112,14 @@ Function Set-ConstructedState($initial_state, $desired_state) {
             @extra_args
     } Catch {
-      Fail-Json -obj $result -message "Failed to move the AD object $($desired_state.name) to $($desired_state.ou) OU: $($_.Exception.Message)"
+      Fail-Json -obj $result -message "Failed to move the AD object $($initial_state.distinguished_name) to $($desired_state.distinguished_name) OU: $($_.Exception.Message)"
```
This changed message seems a bit off in syntax. maybe remove the 'ou'  at the end.

sergioirium replied:
corrected

ShachafGoldstein commented:
This is missing a changelog fragment as well.

Daniel-Sanchez-Fabregas commented on lib/ansible/modules/windows/win_domain_computer.ps1:
```diff
@@ -112,14 +112,14 @@ Function Set-ConstructedState($initial_state, $desired_state) {
     # Move computer to OU
     Try {
-      Get-ADComputer -Identity $desired_state.name |
+      Get-ADComputer -Identity $desired_state.name -credential $credential |
```
~~~ suggestion
Get-ADComputer -Identity $desired_state.name @extra_args |
~~~

Daniel-Sanchez-Fabregas replied:
when domain_username isn't passed as parameter it will fail
`@extra_args` prevents it :
https://github.com/ansible/ansible/blob/bace122177a9fd290fc1ffe7403d613405864b23/lib/ansible/modules/windows/win_domain_computer.ps1#L35-L39

Daniel-Sanchez-Fabregas commented on lib/ansible/modules/windows/win_domain_computer.ps1:
```diff
@@ -147,8 +147,8 @@ Function Add-ConstructedState($desired_state) {
   Try {
-    Get-ADComputer $initial_state.name `
-    | Remove-ADObject `
+    Get-ADComputer -identity $initial_state.name -credential $credential |
```
~~~ suggestion
Get-ADComputer -Identity $initial_state.name @extra_args |
~~~

sergioirium replied:
I've tried it and it also works

sergioirium committed:
Update lib/ansible/modules/windows/win_domain_computer.ps1

Co-Authored-By: Daniel-Sanchez-Fabregas <33929811+Daniel-Sanchez-Fabregas@users.noreply.github.com>

sergioirium committed:
Update win_domain_computer.ps1

changed
"-credential $credential" to "@extra_args" (Line 115 and 150)
corrected exception message (Line 122)

jborean93 committed:
Added changelog fragment, minor code tweak

jborean93 commented:
Added a changelog fragement and slight formatting changes.

jborean93 commented:
Backport PRs
stable-2.8 https://github.com/ansible/ansible/pull/64755
stable-2.9 https://github.com/ansible/ansible/pull/64756
